social-network-api/src/main/java/fs/socialnetworkapi/service/UserService.java

```diff
@@ -48,14 +52,32 @@ public User saveUser(User user) {
   }
 
   public UserDtoOut showUser(Long userId) {
-    User user = findById(userId);
-    return mapper.map(user, UserDtoOut.class);
+    User user = userRepo.getReferenceById(userId);
+
+    user.setUserFollowingCount(getFollowings(userId).size());
+    user.setUserFollowersCount(getFollowers(userId).size());
+    user.setUserTweetCount(getUserPosts(userId, 0, 1000000).size());// need to correct
+
+    return mapper.map(userRepo.save(user), UserDtoOut.class);
+  }
+
+  public List<PostDtoOut> getUserPosts(Long currentUserId, Integer page, Integer size) {
+
+    User user = userRepo.findById(currentUserId)
+      .orElseThrow(() -> new UserNotFoundException(String.format("User with id: %d not found", currentUserId)));
+
+    PageRequest pageRequest = PageRequest.of(page, size, Sort.by(Sort.Direction.DESC, "createdDate"));
+
+    return postRepo.findByUser(user, pageRequest)
+      .stream()
+      .map(p -> mapper.map(p, PostDtoOut.class))
+      .toList();
   }
 
   public UserDtoOut editUser(UserDtoIn userDtoIn) {
     User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
     String email = user.getEmail();
-    User userFromDb = findByEmail(email);
+    User userFromDb = userRepo.findByEmail(email);
```

Comment:
Same problem if the logic would be changed. Better use findByEmail method. It better just for not doing things that you were doing in all class